src/lib/djerba/plugins/tar/swgs/preprocess.py

```diff
@@ -0,0 +1,101 @@
+"""
+The purpose of this file is deal with pre-processing necessary files for the SWGS plugin.
+They're in a separate file because the pre-processing is a little more complex.
+AUTHOR: Aqsa Alam
+"""
+
+# IMPORTS
+import os
+import csv
+import gzip
+import logging
+import pandas as pd
+from djerba.util.logger import logger
+from djerba.sequenza import sequenza_reader
+from djerba.util.subprocess_runner import subprocess_runner
+from djerba.extract.oncokb.annotator import oncokb_annotator
+from shutil import copyfile
+import djerba.plugins.tar.swgs.constants as constants 
+
+class preprocess(logger):
+
+  def __init__(self, tumour_id, oncotree_code, work_dir, log_level=logging.DEBUG, log_path=None):
+
+    # DIRECTORIES
+    self.logger = self.get_logger(log_level, __name__, log_path)
+    self.work_dir = work_dir
+    self.tmp_dir = os.path.join(self.work_dir, 'tmp')
+    if os.path.isdir(self.tmp_dir):
+        self.logger.debug("Using tmp dir {0} for R script wrapper".format(self.tmp_dir))
+    elif os.path.exists(self.tmp_dir):
+        msg = "tmp dir path {0} exists but is not a directory".format(self.tmp_dir)
+        self.logger.error(msg)
+        raise RuntimeError(msg)
+    else:
+        self.logger.debug("Creating tmp dir {0} for R script wrapper".format(self.tmp_dir))
+        os.mkdir(self.tmp_dir)
+    self.r_script_dir = os.environ.get('DJERBA_BASE_DIR') + "/plugins/tar/Rscripts"
+    self.r_script_dir_swgs = os.environ.get('DJERBA_BASE_DIR') + "/plugins/tar/swgs/" 
+    self.data_dir = os.environ.get('DJERBA_BASE_DIR') + "/data/"
+    self.tumour_id = tumour_id
+    self.oncotree_code = oncotree_code
+
+    # RANDOM
+    self.cache_params = None
+
+  # ----------------------- to do all the pre-processing --------------------
+    
+    
+  def run_R_code(self, seg_path):
+    
+    cmd = [
+        'Rscript', self.r_script_dir_swgs + "/process_CNA_data.r",
+        '--basedir', self.r_script_dir,
+        '--outdir', self.work_dir,
+        '--segfile', seg_path,
+        '--genebed', "/.mounts/labs/gsi/modulator/sw/Ubuntu18.04/djerba-0.4.8/lib/python3.10/site-packages/djerba/data/gencode_v33_hg38_genes.bed",
```

Comment:
Hard-coding this path to a specific modulator install of a specific obsolete version of Djerba is very bad -- the modules define environment variables like `DJERBA_ROOT` for install paths, use `os.environ` to access those instead https://docs.python.org/3/library/os.html#os.environ